sled-agent/src/sim/http_entrypoints_pantry.rs

```diff
@@ -280,3 +280,101 @@ async fn detach(
 
     Ok(HttpResponseDeleted())
 }
+
+#[cfg(test)]
+mod tests {
+    use guppy::graph::ExternalSource;
+    use guppy::graph::GitReq;
+    use guppy::graph::PackageGraph;
+    use guppy::MetadataCommand;
+    use serde_json::Value;
+    use std::path::Path;
+
+    fn load_real_api_as_json() -> serde_json::Value {
+        let manifest_path = Path::new(env!("CARGO_MANIFEST_DIR"))
+            .parent()
+            .unwrap()
+            .join("Cargo.toml");
+        let mut cmd = MetadataCommand::new();
+        cmd.manifest_path(&manifest_path);
+        let graph = PackageGraph::from_command(&mut cmd).unwrap();
+        let package = graph
+            .packages()
+            .find(|pkg| pkg.name() == "crucible-pantry-client")
+            .unwrap();
+        let ExternalSource::Git { req, .. } =
+            package.source().parse_external().unwrap()
+        else {
+            panic!("This should be a Git dependency");
+        };
+        let part = match req {
+            GitReq::Branch(inner) => inner,
+            GitReq::Rev(inner) => inner,
+            GitReq::Tag(inner) => inner,
+            GitReq::Default => "main",
+            _ => unreachable!(),
+        };
+        let raw_url = format!(
+            "https://raw.githubusercontent.com/oxidecomputer/crucible/{part}/openapi/crucible-pantry.json",
+        );
+        let raw_json =
```

Comment:
This test will require internet access to work?

Reply:
That's correct, yeah. We could conceivably cache this in some way, maybe in a `build.rs` file? Short of downloading it and checking it in, I'm not sure I see a foolproof way to avoid some network activity though.

Reply:
Yeah, I guess we can't call it the dumbest way if it becomes to smart :)
Agreed though, a better solution might be too heavy to be worth the effort here.